driver.go

```diff
@@ -77,6 +77,10 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 		mc.netConn, err = nd.Dial(mc.cfg.Net, mc.cfg.Addr)
 	}
 	if err != nil {
+		if nerr, ok := err.(net.Error); ok && (nerr.Temporary() || nerr.Timeout()) {
```

Comment:
Don't `|| nerr.Timeout()`.

`nerr.Temporary()` returns True when retry is worthful.
For example, lookup timeout is temporary.

```
func (e *DNSError) Temporary() bool { return e.IsTimeout || e.IsTemporary }
```

Reply:
:+1:

the docs say
```
IsTimeout   bool   // if true, timed out; not all timeouts set this
IsTemporary bool   // if true, error is temporary; not all errors set this; added in Go 1.6
```
so I thought it would be better to check both given `IsTemporary` might not be supported